main.js

```diff
@@ -0,0 +1,43 @@
+const resourcesList = Array.from(
+	document.querySelectorAll(".resources__text-title")
+);
+const resourcesItems = document.querySelectorAll("[data-id]");
+const markerList = Array.from(document.querySelectorAll(".resources__marker"));
+const resourcesText = document.querySelectorAll("[data-texts]");
+const rolling = document.querySelectorAll("shared-hosting__block");
+const arrows = document.querySelectorAll("shared-hosting__arrow");
+
+resourcesList.forEach((text) => {
```

Comment:
Niezla kamehameha https://miro.medium.com/max/1280/0*b_SQxtNbQIKDfn20.png. Trzeba porozbijac to na mniejsze fukncje. Generalnie fajnie by bylo, gdyby funckja nie miala wiecej niz 6 linijek, Czy mniej wiecej cos takiego

```

const addResourceEventListeners = (resource) => {
  resource.addEventListener("click", () => handleResourceClick(resource));
};

const handleResourceClick = (resource) => {
  resource.style.color = "#673de6";

  resourcesList.forEach((anotherResource) => {
    if (resource.dataset.id !== anotherResource.dataset.id)
      return resetResourceColor(resource);
    }

    markerList.forEach((marker) => {
      if (text.dataset.id === marker.dataset.id) {
        marker.classList.remove("resources__marker--none");
        resourcesText.forEach((description) => {
          if (item.dataset.id === description.dataset.texts) {
            description.classList.add("resources__right-md--active");
          } else description.classList.remove("resources__right-md--active");
        });
      } else {
        marker.classList.add("resources__marker--none");
      }
    });
  });
};

const resetResourceColor = (resource) => {
  resource.style.color = "black";
}

resourcesList.forEach(addResourceEventListeners);

```
W tym przykladzie rozbilem tylko czesc, zebys wiedzial o co chodzi. Zwroc uwage na ten zapis 
```
 if (resource.dataset.id !== anotherResource.dataset.id)
      return resetResourceColor(resource);
    }

    markerList.forEach((marker) => {
      if (text.dataset.id === marker.dataset.id) {
        marker.classList.remove("resources__marker--none");
        resourcesText.forEach((description) => {
          if (item.dataset.id === description.dataset.texts) {
            description.classList.add("resources__right-md--active");
          } else description.classList.remove("resources__right-md--active");
        });
      } else {
        marker.classList.add("resources__marker--none");
      }
    });
```
Dzieki temu kamehama jest duzo mniejsza niz tutaj
```
    if (resource.dataset.id === anotherResource.dataset.id)
      markerList.forEach((marker) => {
        if (text.dataset.id === marker.dataset.id) {
          marker.classList.remove("resources__marker--none");
          resourcesText.forEach((description) => {
            if (item.dataset.id === description.dataset.texts) {
              description.classList.add("resources__right-md--active");
            } else description.classList.remove("resources__right-md--active");
          });
        } else {
          marker.classList.add("resources__marker--none");
        }
      });
    } else {
      resetResourceColor(resource);
    }
```